### PR TITLE
added ONBOARDING userstate to render userStatus UI

### LINKED ISSUE #392: Problem in my-site i.e.status page not showing option to choose status when user state: ONBOARDING after signup
https://user-images.githubusercontent.com/107163260/234006823-eedbc5a4-e603-48f7-aee3-341c54d4f563.mp4

If you see video above i.e. https://my.realdevsquad.com/ (my-site)  (`localhost:3443`)  when new user signup and 
`state: ONBOARDING` added in userStatus Collection in database bydefault.

**Problem** 
when new user go to https://my.realdevsquad.com/ (i.e. localhost:3443)  then new user not able to update new state manually because screen is blank (i.e. refer above video) 

**Solution**
do some changes in website-my repo i.e. userStatus related file to pass the condition and see the UI i.e. ACTIVE, OOO, IDLE like before in my-site  https://my.realdevsquad.com/  so that new user can change there state: ONBORDING to any other state. 
image

### PR DESCRIPTION
**Pull request by shubhamsigdar1**
This pr is dependent on pr: https://github.com/Real-Dev-Squad/website-backend/pull/1031

### What is the change?

- change in file `app/components/user-status.js`, `app/components/user-status.hbs`, `app/constants/user-status.js`, `app/styles/user-status.css`, `app/styles/variables.css`
- added element in array currentUserStatus of app/components/user-status.js to render userStatus UI
- make a card component to show the message with user in onboarding state
-

### Before Change Video

https://user-images.githubusercontent.com/107163260/234451815-9e3c1ace-eff9-4583-9758-16fa7d09af7d.mp4

### After Change Video

**Case1: When user directly go to my-site and change the status and then user signup**

https://github.com/Real-Dev-Squad/website-backend/assets/107163260/3f631574-fd2d-4277-8b98-a91ad69ef9f1

**Case2: When user already signup => first change status + then fill join form then changes status**

https://github.com/Real-Dev-Squad/website-backend/assets/107163260/ce5a4306-8325-4279-8c6a-c3b92dccc528

**Responsiveness**

https://user-images.githubusercontent.com/107163260/235363490-5fc6f084-7744-4d74-918e-1294ab76da49.mp4

### CODE REVIEW

File: app/styles/user-status.css [hunk 1]
```diff
@@ -1,3 +1,36 @@
+.onboarding-details {
+  width: 90%;
+  max-width: 450px;
+  border-radius: 10px;
+  background-color: var(--white);
+  box-shadow: 0 0 15px -7px var(--onboarding-card--box-shadow);
```
Comment (RitikJaiswal75) [inline app/styles/user-status.css:6]: please try not to use odd numbers for pixel size

File: app/components/user-status.hbs [hunk 2]
```diff
@@ -5,9 +5,24 @@
...
+          and ask for doubts in the team chat on Discord.
+        </p>
+        <p data-test-details>If you are not a developer, please contact Ankush for next steps.</p>
+        <button class="buttons__active" type="button"  {disabled={{@isStatusUpdating}} {{on "click" (fn
+          this.changeStatus "ACTIVE" )}} data-test-active-button>
+          <span>Change your status to Active</span>
+        </button>
+      </div>
     {{/if}}
+    <h2 data-test-status>{{currentStatus.message}}</h2>
```
Comment (heyrandhir) [inline app/components/user-status.hbs:24]: should this be in the `else`. i.e if the status is `onboarding` then we shouldn't show this. Right ?

**Review by heyrandhir — CHANGES_REQUESTED** [review]

**Review by RitikJaiswal75 — COMMENTED** [review]
Please write tests for it

**Review by RitikJaiswal75 — APPROVED** [review]

**Review by heyrandhir — APPROVED** [review]

**Comment by heyrandhir** [comment]
Approving it for now as this was one off exception. Please write the tests ASAP.

**Pull request merged**